it/v2/endpoints/CreateUkPropertyPeriodSummaryControllerISpec.scala

```diff
@@ -0,0 +1,622 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.endpoints
+
+import com.github.tomakehurst.wiremock.stubbing.StubMapping
+import play.api.http.HeaderNames.ACCEPT
+import play.api.http.Status
+import play.api.http.Status.BAD_REQUEST
+import play.api.libs.json.{JsValue, Json}
+import play.api.libs.ws.{WSRequest, WSResponse}
+import support.V2IntegrationBaseSpec
+import v2.models.errors._
+import v2.stubs.{AuditStub, AuthStub, IfsStub, MtdIdLookupStub}
+
+class CreateUkPropertyPeriodSummaryControllerISpec extends V2IntegrationBaseSpec {
+
+  private val requestBodyJson = Json.parse(
```

Comment:
Can we use a `def` here and pass in fromDate and toDate instead of having huge chunks of Json for the various date scenarios?

Reply:
I can remove the optional values

Reply:
I thought vals by name clearly mentioned what exactly the purpose is!!!